File: app/src/main/java/com/katkov/lolachievements/di/provider/PrefserProvider.java
package com.katkov.lolachievements.di.provider;

import android.content.SharedPreferences;

import com.github.pwittchen.prefser.library.rx2.Prefser;

import javax.inject.Inject;
import javax.inject.Provider;

public final class PrefserProvider implements Provider<Prefser> {

    private final SharedPreferences sharedPreferences;

    @Inject
    public PrefserProvider(SharedPreferences sharedPreferences) {
        this.sharedPreferences = sharedPreferences;
    }

    @Override
    public Prefser get() {
        return new Prefser(sharedPreferences);
    }
}

File: app/src/main/java/com/katkov/lolachievements/di/provider/PrefserProvider.kt
package com.katkov.lolachievements.di.provider

import android.content.SharedPreferences

import com.github.pwittchen.prefser.library.rx2.Prefser

import javax.inject.Inject
import javax.inject.Provider

class PrefserProvider
@Inject
constructor(private val sharedPreferences: SharedPreferences) : Provider<Prefser> {

    override fun get(): Prefser = Prefser(sharedPreferences)
}